sdk/metric/instrument.go

```diff
@@ -53,3 +62,21 @@ func (i *instrumentImpl[N]) Observe(ctx context.Context, val N, attrs ...attribu
 		agg.Aggregate(val, attribute.NewSet(attrs...))
 	}
 }
+
+func (i *instrumentImpl[N]) Add(ctx context.Context, val N, attrs ...attribute.KeyValue) {
+	if err := ctx.Err(); err != nil {
+		return
+	}
+	for _, agg := range i.aggregators {
+		agg.Aggregate(val, attribute.NewSet(attrs...))
+	}
+}
+
+func (i *instrumentImpl[N]) Record(ctx context.Context, val N, attrs ...attribute.KeyValue) {
+	if err := ctx.Err(); err != nil {
```

Comment:
These are equivalent, just a method name difference, right?

```suggestion
func (i *instrumentImpl[N]) Add(ctx context.Context, val N, attrs ...attribute.KeyValue) {
	i.aggregate(ctx, val, attrs)
}

func (i *instrumentImpl[N]) Record(ctx context.Context, val N, attrs ...attribute.KeyValue) {
	i.aggregate(ctx, val, attrs)
}

func (i *instrumentImpl[N]) aggregate(ctx context.Context, val N, attrs []attribute.KeyValue) {
	if err := ctx.Err(); err != nil {
		return
	}
	for _, agg := range i.aggregators {
		agg.Aggregate(val, attribute.NewSet(attrs))
	}
}
```

Reply:
You are right, it's also the same code in Observe().  Refactored the lot of them.